Lib/test/test_sys.py

```diff
@@ -1076,14 +1076,21 @@ class B:
                 class X(Exception):
                     pass
 
-        with test.support.captured_stderr() as stderr, \
-             test.support.swap_attr(sys, 'unraisablehook',
-                                    sys.__unraisablehook__):
-                 expected = self.write_unraisable_exc(
-                     A.B.X(), "msg", "obj");
-        report = stderr.getvalue()
-        testName = 'test_original_unraisablehook_exception_qualname'
-        self.assertIn(f"{testName}.<locals>.A.B.X", report)
+        for moduleName in 'builtins', '__main__', 'some_module':
+            with self.subTest(moduleName=moduleName):
+                A.B.X.__module__ = moduleName
+                with test.support.captured_stderr() as stderr, \
+                     test.support.swap_attr(sys, 'unraisablehook',
+                                            sys.__unraisablehook__):
+                         expected = self.write_unraisable_exc(
+                             A.B.X(), "msg", "obj");
+                report = stderr.getvalue()
+                testName = 'test_original_unraisablehook_exception_qualname'
+                self.assertIn(f"{testName}.<locals>.A.B.X", report)
+                if moduleName in ['builtins', '__main__']:
+                    self.assertNotIn(moduleName, report)
+                else:
+                    self.assertIn(moduleName, report)
```

Comment:
Hum, it's module + ".", no? I suggest to check for moduleName + "." in both code paths. A traceback contains filenames, and path can contain "builtins", but "builtins." is even less likely :-)